cmd/infrakit/base/template.go

```diff
@@ -131,7 +132,18 @@ func TemplateProcessor(plugins func() discovery.Plugins) (*pflag.FlagSet, ToJSON
 				key := strings.TrimSpace(kv[0])
 				val := strings.TrimSpace(kv[1])
 				if key != "" && val != "" {
-					engine.Global(key, val)
+					// Attempt to convert to int and bool types so that template operations
+					// are not only against strings. Only honor "true"/"false" for bool so
+					// that a string like "no" will remain a string.
+					if intVal, err := strconv.Atoi(val); err == nil {
+						engine.Global(key, intVal)
+					} else if val == "true" {
```

Comment:
I see your point about not using the `parseBool` function.  In this case, there's a `strconv.ParseBool` that should do the job.. https://golang.org/pkg/strconv/#ParseBool